app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -471,6 +476,18 @@ public void run() {
         }, 20);
     }
 
+    void uploadImmersiveElapsedTime() {
+        if (startImmersiveTime == 0) {
+            return;
+        }
+
+        long elapsedImmersive = SystemClock.elapsedRealtime() - startImmersiveTime;
+        if (elapsedImmersive > MIN_IMMERSIVE_TIME) {
+            Log.i(LOGTAG, "Send immersive time spent to histogram.");
+            TelemetryWrapper.addImmersiveToHistogram(elapsedImmersive);
+        }
+    }
+
```

Comment:
nit: In other projects I'd recommend to move all this into TelemetryWrapper and just expose start/upload methods and not have any logic for this in the activity.

Reply:
Ok. suggestion for hide the logic into TelemetryWrapper. I will fix it.